src/Scene.cpp

```diff
@@ -284,7 +296,9 @@ void Scene::UpdateMarble(float dx, float dy) {
       const float fz = marble_pos.z() - flag_pos.z();
       if (fx*fx + fz*fz < 6 * marble_rad*marble_rad) {
         final_time = timer;
-        high_scores.Update(cur_level, final_time);
+        if (!scale_marble_flag && !moon_gravity_flag) {
```

Comment:
I suggest turning this into a `bool has_cheated`. Either a variable, or even better, a function.
Alternative name: `bool can_save` or similar.

Basically make this more readable and extendable for the future.